x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc

```diff
@@ -0,0 +1,86 @@
+[[add-session-metadata]]
+=== Add session metadata
+
+beta::[]
+
+++++
+<titleabbrev>add_session_metadata</titleabbrev>
+++++
+
+ifeval::["{beatname_lc}"=="auditbeat"]
+
+The `add_session_metadata` processor enriches process events with additional
+information that users can see using the {security-guide}/session-view.html[Session View] tool in the
+{elastic-sec} platform. 
+
+Here's an example using the `add_session_metadata` processor to enhance process events generated by
+the `auditd` module of {auditbeat}.
+
+[source,yaml]
+-------------------------------------
+auditbeat.modules:
+- module: auditd
+  processors:
+    - add_session_metadata:
+       backend: "auto"
+-------------------------------------
+
+[[add-session-metadata-explained]]
+==== How the add_session_metadata processor works
+
+The `add_session_metadata` processor operates using various backend options, including `auto`, `ebpf`, and `procfs`. 
```

Comment:
We should advise what is necessary in terms of permissions and OS config. Imaging you are a user trying to run this inside a container using least privileges. Do I need to be root? What capabilities are required? Do I need to have debugfs and tracefs mounted? etc.

Reply:
Is this question for @nick-alayil? @mjwolf? Either or both? :-)

Reply:
The initial release of the auditbeat + session view was limited to VMs and bare metal environments. Support for Docker and Kubernetes is planned to follow. @karenzone we should mention this in the documentation.

Reply:
We're careful not to make commitments for future development in format documentation.  I added a note that says "The current release of `add_session_metadata` processor for {auditbeat} limited to virtual machines (VMs) and bare metal environments."  This accurately describes current state while hinting that more enhancements are coming.

Reply:
The multi-threaded comments are getting difficult to follow.   If I'm not mistaken, we don't have an answer to @andrewkroh's comment : 

>"We should advise what is necessary in terms of permissions and OS config. Imaging you are a user trying to run this inside a container using least privileges. Do I need to be root? What capabilities are required? Do I need to have debugfs and tracefs mounted? etc."

This is outside my area of expertise, and I'm not sure how to resolve this comment.  Can somebody help? 
Bonus points of you answer the question in the form of a github suggestion so that your intent for content is 100% clear.

Reply:
```suggestion
==== Backends
* `auto` is the recommended setting, and will attempt to use `ebpf` first, falling back to `procfs` if necessary, ensuring compatibility even on systems without `ebpf` support.
* `ebpf` will collect process information with eBPF. This backend requires a system with Linux kernel 5.10.16 or above, kernel support for eBPF enabled, and auditbeat running as superuser.
* `procfs` will collect process information with the proc filesystem. This is compatible with older systems that may not support ebpf. To gather complete process info, auditbeat requires permissions to read all process data in procfs; for example, run as a superuser or have the `SYS_PTRACE` capability. 

==== Containers
If running auditbeat in a container, the container must run in the host's PID namespace. With the `auto` or `ebpf` backend, these host directories must also be mounted to the same path within the container: `/sys/kernel/debug`, `/sys/fs/bpf` 
```

This will probably need some editing, but this should be the information needed